third_party/BUILD

```diff
@@ -517,14 +517,36 @@ java_import(
     ],
 )
 
+UNNECESSARY_DYNAMIC_LIBRARIES = select({
+    "//src/conditions:windows": "*.so *.jnilib",
+    "//src/conditions:darwin": "*.so *.dll",
+    "//src/conditions:darwin_x86_64": "*.so *.dll",
+    "//src/conditions:linux_x86_64": "*.jnilib *.dll",
+    # The .so file is an x86 one, so we can just remove it the CPU is not x86
+    "//src/conditions:arm": "*.so *.jnilib *.dll",
+    "//src/conditions:linux_aarch64": "*.so *.jnilib *.dll",
+    # Play it safe -- better have a big binary than a slow binary
+    # zip -d does require an argument. Supply something bogus.
+    "//conditions:default": "*.bogusextension",
+})
+
+# Remove native libraries that are for a platform different from the one we are
+# building Bazel for
+genrule(
+    name = "filter_netty_dynamic_libs",
+    srcs = ["netty_tcnative/netty-tcnative-boringssl-static-2.0.24.Final.jar"],
+    outs = ["netty_tcnative/netty-tcnative-filtered.jar"],
+    cmd = "cp $< $@ && zip -qd $@ " + UNNECESSARY_DYNAMIC_LIBRARIES,
```

Comment:
The zip file is non-deterministic, isn't it? In other words: does the bazel determinism test pass?

Reply:
It does! It just removes entries from the zip and does not change the time stamps of existing ones. Verified by `unzip -l bazel-genfiles/third_party/netty_tcnative/netty-tcnative-filtered.jar`: timestamps are 2019 April 2. The determinism test also passes.